mlflow/models/evaluation/base.py

```diff
@@ -151,6 +156,178 @@ def __str__(self):
+                    for name in allowed_kwargs_params.keys()
+                ],
+            ]
+        )
+        call_method = gen_ai_with_llm_judge_call_method
+
+    else:
+        allowed_kwargs_names = [
+            param_name
+            for param_name in inspect.signature(eval_fn).parameters.keys()
+            if param_name not in ["predictions", "metrics", "inputs"]
+        ]
+
+        def genai_call_method(
+            self,
+            *,
+            predictions: Union[pd.Series, str, list],
+            inputs: Union[pd.Series, str, list],
+            metrics: Optional[Dict[str, MetricValue]] = None,
+            return_only_scores: bool = False,
+            **kwargs,
+        ) -> Union[MetricValue, List[str], List[float]]:
+            """
+            Evaluate the metric on the given inputs and predictions.
+            Note: only keyword arguments are supported.
+
+            Args:
+                predictions: predictions made by the model.
+                inputs: inputs used to make the predictions.
+                metrics: metrics calculated by the default evaluator.
+                return_only_scores: If True, return only the scores from the metric.
+                    Otherwise, return the full MetricValue object. Default to False.
+                kwargs: additional arguments used to compute the metric.
+
+            Returns:
+                If return_only_scores is True, return the scores from the metric.
+                Otherwise, return the full MetricValue object.
+            """
+
+            if missed_kwargs := set(allowed_kwargs_names) - set(kwargs.keys()):
+                raise MlflowException.invalid_parameter_value(
+                    f"Missing required arguments: {missed_kwargs}",
+                )
+            if extra_kwargs := set(kwargs.keys()) - set(allowed_kwargs_names):
+                raise MlflowException.invalid_parameter_value(
+                    f"Unexpected arguments: {extra_kwargs}",
+                )
+            result = self.eval_fn(
+                _convert_val_to_pd_Series(predictions, "predictions"),
+                metrics or {},
+                _convert_val_to_pd_Series(inputs, "inputs"),
+                # Note: based on https://github.com/mlflow/mlflow/blob/4fef77afdbe4d76302cb0b1aad2bd72b5cde64e9/mlflow/metrics/genai/genai_metric.py#L49-L53
+                # the extra params passed https://github.com/mlflow/mlflow/blob/4fef77afdbe4d76302cb0b1aad2bd72b5cde64e9/mlflow/metrics/genai/genai_metric.py#L513
+                # should always be pandas Series
+                *[
+                    _convert_val_to_pd_Series(kwargs[arg_name], arg_name)
+                    for arg_name in allowed_kwargs_names
+                ],
+            )
+            if return_only_scores:
+                return result.scores
+            return result
+
+        genai_call_method.__signature__ = Signature(
```

Comment:
what would happen if we didn't update the signature?

Reply:
Then the signature only contains kwargs, users don't know what kwargs they should pass

Reply:
> users don't know what kwargs they should pass

I see, how do users check kwargs?

Reply:
Yes, `inspect.signature` works, also `help` works (see the PR description)